### PR TITLE
Modified analyzer to perform matching brace analysis at bit more efficient

### LINKED ISSUE #734: Rewrite SA1513ClosingCurlyBracketMustBeFollowedByBlankLine.IsOnSameLineAsOpeningBrace for performance
The line to `currentToken = currentToken.GetPreviousToken();` in this method is currently responsible for over 12% of the overall cost of running the currently available StyleCopAnalyzers analyzers. This issue was found while investigating #732.

### PR DESCRIPTION
**Pull request by vweijsters**
Fixes #734 

Haven't done the performance analysis, but the algorithm is much more efficient, so this should be good.

@sharwell How did you do the performance measurements?

### CODE REVIEW

**Comment by sharwell** [comment]
I use the Visual Studio profiler. I initially start without profiling, and then after I load a solution I enable profiling and type for a while.

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1513ClosingCurlyBracketMustBeFollowedByBlankLine.cs [hunk 1]
```diff
@@ -66,204 +69,201 @@ public override void Initialize(AnalysisContext context)
             context.RegisterSyntaxTreeActionHonorExclusions(this.HandleSyntaxTreeAction);
         }
 
-        private void HandleSyntaxTreeAction(SyntaxTreeAnalysisContext context)
+        private async void HandleSyntaxTreeAction(SyntaxTreeAnalysisContext context)
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1513ClosingCurlyBracketMustBeFollowedByBlankLine.cs]: :exclamation: `async void` should be avoided.

Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1513ClosingCurlyBracketMustBeFollowedByBlankLine.cs]: Good point, I fixed it.

**Comment by sharwell** [comment]
Verified the improvement. There are now 8 rules slower than this rule, and it's only 2.17% of the total time spent in StyleCopAnalyzers during a long profiling run of working on the Roslyn project.

**Pull request merged**